charm-slurmd/src/charm.py

```diff
@@ -183,6 +195,37 @@ def _on_slurmctld_available(self, event):
         self._on_set_partition_info_on_app_relation_data(event)
         self._check_status()
 
+        # check etcd for hostnames
+        self.on.check_etcd.emit()
+
+    def _on_check_etcd(self, event):
+        """Check if node is accounted for.
+
+        Check if slurmctld accounted for this node's inventory for the first
+        time, if so, emit slurmctld_started event, so the node can start the
+        daemon.
+        """
+
+        host = self._slurmd.slurmctld_address
+        logger.debug(f"## Connecting to etcd3 in {host}")
+        client = Etcd3Client(host=host, port=2379, api_path="/v3/")
```

Comment:
Should port be pulled from environment or config or something? Is it safe to assume etcd is *always* running on port 2379

Reply:
etcd is an internal thing only, it should not be used for end-users, and is used for a very specific purpose. I don't see a reason to need to change the port.

Thoughts @jamesbeedy ?